autofaiss/external/scores.py

```diff
@@ -28,15 +27,11 @@ def compute_fast_metrics(
     size_bytes = get_index_size(index)
     infos["size in bytes"] = size_bytes
 
-    if isinstance(embeddings_path, (list, str)):
+    if isinstance(embedding_reader, EmbeddingReader):
         # pylint: disable=bare-except
-        query_embeddings, _ = next(
-            read_embeddings(
-                embeddings_path, file_format=file_format, embedding_column_name=embedding_column_name, verbose=False
-            )
-        )
+        query_embeddings, _ = next(embedding_reader(start=0, end=query_max, batch_size=query_max))
     else:
-        query_embeddings = embeddings_path
+        query_embeddings = embedding_reader
 
     query_embeddings = query_embeddings[:query_max]
```

Comment:
If I understand correctly, this line is not necessary if we enter into the first `if` because `query_max` has been take into account by `embedding_reader` thanks to `end` and `batch_size`.

Reply:
That's true indeed but it doesn't cost much here

Reply:
I fixed it anyway